src/main/java/io/reactivex/observers/BaseTestConsumer.java

```diff
@@ -565,9 +565,14 @@ public final U assertValuesOnly(T... values) {
     }
 
     /**
-     * Assert that the TestObserver/TestSubscriber received only the specified values in any order.
-     * <p>This helps asserting when the order of the values is not guaranteed, i.e., when merging
+     * Assert that the TestObserver/TestSubscriber received only items that are in the specified
+     * collection as well irrespective of the order they were received.
+     * <p>
+     * This helps asserting when the order of the values is not guaranteed, i.e., when merging
      * asynchronous streams.
+     * <p>
+     * To ensure that only the expected items have been received, no more and no less, in any order,
+     * apply {@link #assertValueCount(int)} with {@code expected.size()}.
```

Comment:
As-is this method "ignores" duplicates, while `assertValueCount` doesn't. If a user wishes to assert that all values were received while allowing for duplicates, they could to `.distinct().test()`. Is this the preferred approach, and if so, is it worth documenting that too? (Maybe have an example of that in the test below?)

Reply:
The aim is to provide some typical set of assertions with `TestObserver`. You can always modulate the input flow or simply enumerate contents of the `TestObserver` manually. I usually try to minimize the cases where additional operators are mentioned in the javadocs because it triggers "what about when using X" type of comments.